protocol/contracts/templegold/DaiGoldAuction.sol

```diff
@@ -0,0 +1,342 @@
+        depositors[msg.sender][epochIdCache] += amount;
+
+        EpochInfo storage info = epochs[epochIdCache];
+        info.totalBidTokenAmount += amount;
+        emit Deposit(msg.sender, epochIdCache, amount);
+    }
+
+    /**
+     * @notice Claim share of Temple Gold for epoch
+     * Can only claim for past epochs, not current auction epoch.
+     * @param epochId Id of epoch
+     */
+    function claim(uint256 epochId) external virtual override {
+        /// @notice cannot claim for current live epoch
+        EpochInfo storage info = epochs[epochId];
+        if (!info.hasEnded()) { revert CannotClaim(epochId); }
+        /// @dev epochId could be invalid. eg epochId > _currentEpochId
+        if (info.startTime == 0) { revert InvalidEpoch(); }
+        if (claimed[msg.sender][epochId]) { revert AlreadyClaimed(); }
+
+        uint256 bidTokenAmount = depositors[msg.sender][epochId];
+        if (bidTokenAmount == 0) { revert CommonEventsAndErrors.ExpectedNonZero(); }
+        claimed[msg.sender][epochId] = true;
+        uint256 claimAmount = bidTokenAmount.mulDivRound(info.totalAuctionTokenAmount, info.totalBidTokenAmount, false);
+        claimedAmount[msg.sender][epochId] = claimAmount;
+        templeGold.safeTransfer(msg.sender, claimAmount);
+        emit Claim(msg.sender, epochId, bidTokenAmount, claimAmount);
+    }
+
+    /**
+     * @notice Checkpoint total Temple Gold distributed for next epoch.
+     * Can only be called by Temple Gold contract or elevated access (for rebalancing purpose)
+     * @param amount Amount of Temple Gold to checkpoint
+     */
+    function notifyDistribution(uint256 amount) external override {
+        if (msg.sender != address(templeGold) && !isElevatedAccess(msg.sender, msg.sig)) { revert CommonEventsAndErrors.InvalidAccess(); }
+        /// @notice Temple Gold contract mints TGLD amount to contract before calling `notifyDistribution`
+        nextAuctionGoldAmount += amount;
+        emit GoldDistributionNotified(amount, block.timestamp);
+    }
+    
+    /**
+     * @notice Get Temple Gold supply for epoch
+     * @param epochId Epoch Id
+     * @return Temple Gold supply
+     */
+    function epochGoldSupply(uint256 epochId) external view override returns (uint256) {
+        return epochs[epochId].totalAuctionTokenAmount;
+    }
+
+    /**
+     * @notice Get current epoch
+     * @return Epoch Id
+     */
+    function currentEpoch() external view override returns (uint256) {
+        return _currentEpochId;
+    }
+
+    /**
+     * @notice Get next epoch
+     * @return Next epoch Id
+     */
+    function nextEpoch() public view override returns (uint256) {
+        return _currentEpochId + 1;
+    }
+
+    /**
+     * @notice Check if current epoch ended. That is, if current auction ended.
+     * @return Bool if epoch ended
+     */
+    function isCurrentEpochEnded() external view override returns (bool) {
+        return epochs[_currentEpochId].hasEnded();
```

Comment:
nit:

this returns `true` when an epoch is deleted with `recoverToken()`.
In a way it is correct, because next time `startAuction()` is called, the `currentEpochId` is incremented. 
But just so you are aware of it

Reply:
Noted